app/src/main/java/com/amaze/filemanager/ui/icons/IconHolder.java

```diff
@@ -100,29 +100,20 @@ private synchronized void processResult(LoadResult result) {
             final String filePath =(result.fso);
 
             synchronized (mAppIcons) {
-
                 mAppIcons.put(filePath, result.result);
             }
 
             // find the request for it
             synchronized (mRequests) {
-
-                for (Map.Entry<ImageView, String> entry : mRequests.entrySet()) {
```

Comment:
Here the code the same to what you wrote, you just extended it. Read [here](https://docs.oracle.com/javase/1.5.0/docs/guide/language/foreach.html).